hostinfo/fileperms.lua

```diff
@@ -0,0 +1,84 @@
+--[[
+Copyright 2015 Rackspace
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS-IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+]]--
+local table = require('table')
+local los = require('los')
+local async = require('async')
+local fs = require('fs')
+local execFileToBuffers = require('./misc').execFileToBuffers
+
+--[[ Check file permissions ]]--
+local HostInfo = require('./base').HostInfo
+local Info = HostInfo:extend()
+function Info:initialize()
+    HostInfo.initialize(self)
+end
+
+function Info:run(callback)
+    if los.type() ~= 'linux' then
+        self._error = 'Unsupported OS for file permissions'
+        callback()
+        return
+    end
+
+    local fileList = {
+        '/etc/grub.conf',
+        '/boot/grub/grub.cfg',
+        '/etc/passwd',
+        '/etc/shadow',
+        '/etc/hosts.allow',
+        '/etc/hosts.deny',
+        '/etc/anacrontab',
+        '/etc/crontab',
+        '/etc/cron.hourly',
+        '/etc/cron.daily',
+        '/etc/cron.weekly',
+        '/etc/cron.monthly',
+        '/etc/cron.d',
+        '/etc/ssh/sshd_config',
+        '/etc/gshadow',
+        '/etc/group',
+        '/etc/login.defs',
+        '/var/run/php-fpm.sock'
+    }
+
+    async.forEachSeries(fileList, function(file, cb)
```

Comment:
It would be much simpler, faster, and more portable to use the stat operation in libuv instead of shelling out to the `stat` command.

In luvit, the builtin `fs` module can be used.

```lua
fs.stat(path, function (err, stat)
  if err then return callback(err) end -- handle error somehow
  -- Use bitwise and to read only the last 3 octal digits.
  local mode = bit.band(stat.mode, 511) -- 511 is 0777 in octal
  -- Do something with result
end
```